Automattic/wp-calypso
# Reader Lists: use Intl.Collator to offer locale-aware sorting of sidebar

#### Changes proposed in this Pull Request

In https://github.com/Automattic/wp-calypso/pull/47537#pullrequestreview-533511171, @blowery helpfully suggested that we should use `Intl.Collator()` to sort lists in the sidebar. 

Some languages have a different sort order (e.g. Swedish) and it's good to respect that rather than forcing English A-Z sorts on to everyone. For example, the å character might appear among other 'a' characters in an English sort, but in Swedish it should appear at the end of the alphabet after z. 

#### Testing instructions


Rename one of your lists to begin with "å".

In http://calypso.localhost:3000/read, check that your lists are sorted by title in the sidebar, with your "å" list appearing in the same position as "a".

Change your interface language to Svenska (sv) in your Account Settings, and check that the "å" list now appears at the end of your lists, after "z".

<img width="281" alt="Screen Shot 2020-11-19 at 17 14 52" src="https://user-images.githubusercontent.com/17325/99620622-c2622f00-2a8a-11eb-884c-176997b9789f.png">
